### ISSUE TITLE
Sandboxing

### ISSUE THREAD

**Issue by johanneskoester** [bioconda/bioconda-buildkite#1]
The buildkite agent does not restrict access to the system it is running on. Hence, we **need** to sandbox our build process in order to avoid security issues especially with pull requests.

* For OSX, we should use [sandbox-exec](https://www.davd.eu/os-x-run-any-command-in-a-sandbox/) as a [command hook](https://buildkite.com/docs/agent/hooks).
* For Linux, we should use [firejail](https://firejail.wordpress.com/).

**Comment by bgruening** [comment]
I'm running a VM in the cloud and don't care that much about it. We could also run buildkite in Docker and mount /var/lib/docker/docker.sock into it. This should also give us some more isolation.

**Comment by johanneskoester** [comment]
Yes indeed. But both things won't help on Mac OS, right? Also, in theory a build script could attack your VM and insert some evil code that affects subsequent builds. Here, Docker would help as well of course.

**Comment by aldrinalberto** [comment]
https://youtu.be/B0Q6RqOWVqI
